Index: Comet.cpp
===================================================================
--- Comet.cpp	(revision 131)
+++ Comet.cpp	(working copy)
@@ -233,6 +233,10 @@
          {
             sscanf(szParamVal, "%512s", staticParams.databaseInfo.szDatabase);
          }
+         if (!strcmp(szParamName, "decoy-prefix"))
+         {
+            sscanf(szParamVal, "%200s", g_staticParams.szDecoyPrefix);
+         }
          else if (!strcmp(szParamName, "nucleotide_reading_frame"))
          {
             sscanf(szParamVal, "%d", &(staticParams.options.iWhichReadingFrame));
Index: CometSearch/CometData.h
===================================================================
--- CometSearch/CometData.h	(revision 131)
+++ CometSearch/CometData.h	(working copy)
@@ -559,6 +559,7 @@
    char            szIonSeries[200];   // used for .out files
    char            szDisplayLine[200]; // used for .out files
    char            szMod[280];         // used for .out files
+   char            szDecoyPrefix[200]; // used for prefix to indicate decoys
    int             iElapseTime;
    char            szDate[28];
    Options         options;
Index: CometSearch/CometPostAnalysis.cpp
===================================================================
--- CometSearch/CometPostAnalysis.cpp	(revision 131)
+++ CometSearch/CometPostAnalysis.cpp	(working copy)
@@ -34,7 +34,7 @@
                                      int maxNumThreads)
 {
    if (!g_staticParams.options.bOutputSqtStream)
-      printf(" - Perform post-search analysis\n");
+      logout(" - Perform post-search analysis\n");
 
    // Create the thread pool containing g_staticParams.options.iNumThreads,
    // each hanging around and sleeping until asked to do a post analysis.
@@ -669,7 +669,7 @@
                   }
                   else
                   {
-                     printf(" Error - XCORR DECOY: dFragMass %f, iFragMass %d, ArraySize %d, InputMass %f, scan %d, z %d\n",
+                     logout(" Error - XCORR DECOY: dFragMass %f, iFragMass %d, ArraySize %d, InputMass %f, scan %d, z %d\n",
                            dFragmentIonMass, 
                            iFragmentIonMass,
                            pQuery->_spectrumInfoInternal.iArraySize, 
@@ -688,7 +688,7 @@
                   }
                   else
                   {
-                     printf(" Error - XCORR DECOY: dFragMass %f, iFragMass %d, ArraySize %d, InputMass %f, scan %d, z %d\n",
+                     logout(" Error - XCORR DECOY: dFragMass %f, iFragMass %d, ArraySize %d, InputMass %f, scan %d, z %d\n",
                            dFragmentIonMass,
                            iFragmentIonMass,
                            pQuery->_spectrumInfoInternal.iArraySize,
Index: CometSearch/CometPreprocess.cpp
===================================================================
--- CometSearch/CometPreprocess.cpp	(revision 131)
+++ CometSearch/CometPreprocess.cpp	(working copy)
@@ -235,21 +235,21 @@
    pdTempRawData = (double *)calloc((size_t)pScoring->_spectrumInfoInternal.iArraySize, (size_t)sizeof(double));
    if (pdTempRawData == NULL)
    {
-      fprintf(stderr, " Error - calloc(pdTempRawData[%d]).\n\n", pScoring->_spectrumInfoInternal.iArraySize);
+      logerr( " Error - calloc(pdTempRawData[%d]).\n\n", pScoring->_spectrumInfoInternal.iArraySize);
       exit(1);
    }
 
    pdTmpFastXcorrData = (double *)calloc((size_t)pScoring->_spectrumInfoInternal.iArraySize, (size_t)sizeof(double));
    if (pdTmpFastXcorrData == NULL)
    {
-      fprintf(stderr, " Error - calloc(pdTmpFastXcorrData[%d]).\n\n", pScoring->_spectrumInfoInternal.iArraySize);
+      logerr( " Error - calloc(pdTmpFastXcorrData[%d]).\n\n", pScoring->_spectrumInfoInternal.iArraySize);
       exit(1);
    }
 
    pScoring->pfFastXcorrData = (float *)calloc((size_t)pScoring->_spectrumInfoInternal.iArraySize, (size_t)sizeof(float));
    if (pScoring->pfFastXcorrData == NULL)
    {
-      fprintf(stderr, " Error - calloc(pfFastXcorrData[%d]).\n\n", pScoring->_spectrumInfoInternal.iArraySize);
+      logerr( " Error - calloc(pfFastXcorrData[%d]).\n\n", pScoring->_spectrumInfoInternal.iArraySize);
       exit(1);
    }
 
@@ -261,7 +261,7 @@
       pScoring->pfFastXcorrDataNL = (float *)calloc((size_t)pScoring->_spectrumInfoInternal.iArraySize, (size_t)sizeof(float));
       if (pScoring->pfFastXcorrDataNL == NULL)
       {
-         fprintf(stderr, " Error - calloc(pfFastXcorrDataNL[%d]).\n", pScoring->_spectrumInfoInternal.iArraySize);
+         logerr( " Error - calloc(pfFastXcorrDataNL[%d]).\n", pScoring->_spectrumInfoInternal.iArraySize);
          exit(1);
       }
    }
@@ -350,7 +350,7 @@
       pScoring->pSparseFastXcorrData = (SparseMatrix *)calloc((size_t)pScoring->iFastXcorrData, (size_t)sizeof(SparseMatrix));
       if (pScoring->pSparseFastXcorrData == NULL)
       {
-         fprintf(stderr, " Error - calloc(pScoring->pSparseFastXcorrData[%d]).\n\n", pScoring->iFastXcorrData);
+         logerr( " Error - calloc(pScoring->pSparseFastXcorrData[%d]).\n\n", pScoring->iFastXcorrData);
          exit(1);
       }
       pScoring->pSparseFastXcorrData[0].bin=0;
@@ -379,7 +379,7 @@
          pScoring->pSparseFastXcorrDataNL = (SparseMatrix *)calloc((size_t)pScoring->iFastXcorrDataNL, (size_t)sizeof(SparseMatrix));
          if (pScoring->pSparseFastXcorrDataNL == NULL)
          {
-            fprintf(stderr, " Error - calloc(pScoring->pSparseFastXcorrDataNL[%d]).\n\n", pScoring->iFastXcorrDataNL);
+            logerr( " Error - calloc(pScoring->pSparseFastXcorrDataNL[%d]).\n\n", pScoring->iFastXcorrDataNL);
             exit(1);
          }
          pScoring->pSparseFastXcorrDataNL[0].bin=0;
@@ -433,7 +433,7 @@
       pScoring->pSparseSpScoreData = (SparseMatrix *)calloc((size_t)NUM_SP_IONS, (size_t)sizeof(SparseMatrix));
       if (pScoring->pSparseSpScoreData == NULL)
       {
-         fprintf(stderr, " Error - calloc(pScoring->pSparseSpScoreData[%d]).\n\n", pScoring->iSpScoreData);
+         logerr( " Error - calloc(pScoring->pSparseSpScoreData[%d]).\n\n", pScoring->iSpScoreData);
          exit(1);
       }
       pScoring->iSpScoreData=0;
@@ -452,7 +452,7 @@
       pScoring->pfSpScoreData = (float *)calloc((size_t)pScoring->_spectrumInfoInternal.iArraySize, (size_t)sizeof(float ));
       if (pScoring->pfSpScoreData == NULL)
       {
-         fprintf(stderr, " Error - calloc(pfSpScoreData[%d])\n", pScoring->_spectrumInfoInternal.iArraySize);
+         logerr( " Error - calloc(pfSpScoreData[%d])\n", pScoring->_spectrumInfoInternal.iArraySize);
          exit(1);
       }
 
@@ -778,7 +778,7 @@
    }
    else  // Should not get here.
    {
-      printf(" Error - iIsotopeError=%d\n\n", g_staticParams.tolerances.iIsotopeError);
+      logout(" Error - iIsotopeError=%d\n\n", g_staticParams.tolerances.iIsotopeError);
       exit(1);
    }
 }
@@ -800,7 +800,7 @@
    pPre->pdCorrelationData = (double *)calloc((size_t)pScoring->_spectrumInfoInternal.iArraySize, (size_t)sizeof(double));
    if (pPre->pdCorrelationData == NULL)
    {
-      fprintf(stderr, " Error - calloc(pdCorrelationData[%d]).\n\n", pScoring->_spectrumInfoInternal.iArraySize);
+      logerr( " Error - calloc(pdCorrelationData[%d]).\n\n", pScoring->_spectrumInfoInternal.iArraySize);
       exit(1);
    }
 
@@ -962,7 +962,7 @@
 
    if (pdSmoothedSpectrum == NULL)
    {
-      fprintf(stderr, " Error - calloc(pdSmoothedSpectrum[%d]).\n", iArraySize);
+      logerr( " Error - calloc(pdSmoothedSpectrum[%d]).\n", iArraySize);
       exit(1);
    }
 
@@ -993,7 +993,7 @@
    pdPeakExtracted = (double *)calloc((size_t)iArraySize, (size_t)sizeof(double));
    if (pdPeakExtracted == NULL)
    {
-      fprintf(stderr, " Error - calloc(pdPeakExtracted[%d]).\n", iArraySize);
+      logerr( " Error - calloc(pdPeakExtracted[%d]).\n", iArraySize);
       exit(1);
    }
 
Index: CometSearch/CometSearch.cpp
===================================================================
--- CometSearch/CometSearch.cpp	(revision 131)
+++ CometSearch/CometSearch.cpp	(working copy)
@@ -57,7 +57,7 @@
 
    if ((fptr=fopen(g_staticParams.databaseInfo.szDatabase, "rb")) == NULL)
    {
-       fprintf(stderr, " Error - cannot read database file \"%s\".\n\n", g_staticParams.databaseInfo.szDatabase);
+       logerr( " Error - cannot read database file \"%s\".\n\n", g_staticParams.databaseInfo.szDatabase);
        exit(1);
    }
 
@@ -70,7 +70,7 @@
 
    if (!g_staticParams.options.bOutputSqtStream)
    {
-      printf(" - Search progress: ");
+      logout(" - Search progress: ");
       fflush(stdout);
    }
 
@@ -86,13 +86,13 @@
       // Expect a '>' for sequence header line.
       if (iTmpCh != '>')
       {
-         fprintf(stderr, "\n\n Error - database file, expecting definition line here.\n");
+         logerr( "\n\n Error - database file, expecting definition line here.\n");
          fgets(szBuf, SIZE_BUF, fptr);
-         fprintf(stderr, "%c%s", iTmpCh, szBuf);
+         logerr( "%c%s", iTmpCh, szBuf);
          fgets(szBuf, SIZE_BUF, fptr);
-         fprintf(stderr, "%s", szBuf);
+         logerr( "%s", szBuf);
          fgets(szBuf, SIZE_BUF, fptr);
-         fprintf(stderr, "%s", szBuf);
+         logerr( "%s", szBuf);
          exit(1);
       } 
 
@@ -126,9 +126,9 @@
             && !(g_staticParams.databaseInfo.iTotalNumProteins%200))
       {
          lCurrPos = ftell(fptr);
-         printf("%3d%%", (int)(100.0 * (double)lCurrPos/(double)lEndPos));
+         logout("%3d%%", (int)(100.0 * (double)lCurrPos/(double)lEndPos));
          fflush(stdout);
-         printf("\b\b\b\b");
+         logout("\b\b\b\b");
       }
 
       // Now search sequence entry; add threading here so that
@@ -144,7 +144,7 @@
 
    if (!g_staticParams.options.bOutputSqtStream)
    {
-      printf(" 100%%\n");
+      logout(" 100%%\n");
    }
 }
 
@@ -215,7 +215,7 @@
          pszTemp=(char *)malloc(seqSize);
          if (pszTemp == NULL)
          {
-            fprintf(stderr, " Error - malloc(szTemp[%d])\n",seqSize);
+            logerr( " Error - malloc(szTemp[%d])\n",seqSize);
             exit(1);
          }
 
@@ -373,7 +373,7 @@
                      // Now get the set of binned fragment ions once to compare this peptide against all matching spectra.
                      if ((pbDuplFragment = (bool*)malloc(g_pvQuery.at(iWhichQuery)->_spectrumInfoInternal.iArraySize * (size_t)sizeof(bool)))==NULL)
                      {
-                        fprintf(stderr, " Error - malloc pbDuplFragments; iWhichQuery = %d\n\n", iWhichQuery);
+                        logerr( " Error - malloc pbDuplFragments; iWhichQuery = %d\n\n", iWhichQuery);
                         exit(1);
                      }
 
@@ -414,10 +414,10 @@
                      if (g_staticParams.options.iDecoySearch)
                      {
 #ifdef _WIN32
-                        _snprintf(szDecoyProteinName, WIDTH_REFERENCE, "DECOY_%s", szProteinName);
+                        _snprintf(szDecoyProteinName, WIDTH_REFERENCE, "%s%s", g_staticParams.szDecoyPrefix, szProteinName);
                         szDecoyProteinName[WIDTH_REFERENCE-1]=0;  // _snprintf does not guarantee null termination
 #else
-                        snprintf(szDecoyProteinName, WIDTH_REFERENCE, "DECOY_%s", szProteinName);
+                        snprintf(szDecoyProteinName, WIDTH_REFERENCE, "%s%s", g_staticParams.szDecoyPrefix, szProteinName);
 #endif
                         // Generate reverse peptide.  Keep prev and next AA in szDecoyPeptide string.
                         // So actual reverse peptide starts at position 1 and ends at len-2 (as len-1
@@ -780,7 +780,7 @@
       }
       else
       {
-         fprintf(stderr, " Error - iIsotopeError=%d, should not be here!\n\n", g_staticParams.tolerances.iIsotopeError);
+         logerr( " Error - iIsotopeError=%d, should not be here!\n\n", g_staticParams.tolerances.iIsotopeError);
          exit(1);
       }
    }
@@ -811,10 +811,10 @@
             pTmp=(char *)realloc(_proteinInfo.pszProteinSeq, ii+100);
             if (pTmp == NULL)
             {
-               fprintf(stderr, " Error realloc(szProteinSeq) ... size=%d\n", ii);
-               fprintf(stderr, " A sequence entry is larger than your system can handle.\n");
-               fprintf(stderr, " Either add more memory or edit the database and divide\n");
-               fprintf(stderr, " the sequence into multiple, overlapping, smaller entries.\n\n");
+               logerr( " Error realloc(szProteinSeq) ... size=%d\n", ii);
+               logerr( " A sequence entry is larger than your system can handle.\n");
+               logerr( " Either add more memory or edit the database and divide\n");
+               logerr( " the sequence into multiple, overlapping, smaller entries.\n\n");
                exit(1);
             }
 
@@ -841,10 +841,10 @@
             pTmp=(char *)realloc(_proteinInfo.pszProteinSeq, ii+100);
             if (pTmp == NULL)
             {
-               fprintf(stderr, " Error realloc(szProteinSeq) ... size=%d\n", ii);
-               fprintf(stderr, " A sequence entry is larger than your system can handle.\n");
-               fprintf(stderr, " Either add more memory or edit the database and divide\n");
-               fprintf(stderr, " the sequence into multiple, overlapping, smaller entries.\n\n");
+               logerr( " Error realloc(szProteinSeq) ... size=%d\n", ii);
+               logerr( " A sequence entry is larger than your system can handle.\n");
+               logerr( " Either add more memory or edit the database and divide\n");
+               logerr( " the sequence into multiple, overlapping, smaller entries.\n\n");
                exit(1);
             }
 
@@ -2227,7 +2227,7 @@
             // now get the set of binned fragment ions once for all matching peptides
             if ((pbDuplFragment = (bool*)malloc(g_pvQuery.at(iWhichQuery)->_spectrumInfoInternal.iArraySize * (size_t)sizeof(bool)))==NULL)
             {
-               fprintf(stderr, " Error - malloc pbDuplFragments; iWhichQuery = %d\n\n", iWhichQuery);
+               logerr( " Error - malloc pbDuplFragments; iWhichQuery = %d\n\n", iWhichQuery);
                exit(1);
             }
 
Index: CometSearch/CometSearchManager.cpp
===================================================================
--- CometSearch/CometSearchManager.cpp	(revision 131)
+++ CometSearch/CometSearchManager.cpp	(working copy)
@@ -124,8 +124,8 @@
 
        if (!g_staticParams.options.bOutputSqtStream)
        {
-          printf(" Comet version \"%s\"\n", comet_version);
-          printf(" Search start:  %s\n", g_staticParams.szDate);
+          logout(" Comet version \"%s\"\n", comet_version);
+          logout(" Search start:  %s\n", g_staticParams.szDate);
        }
 
        int iFirstScan = g_staticParams.inputFile.iFirstScan;             // First scan to search specified by user.
@@ -156,13 +156,17 @@
        if (g_staticParams.options.bOutputSqtFile)
        {
           if (iAnalysisType == AnalysisType_EntireFile)
+             #ifdef CRUX
+             sprintf(szOutputSQT, "%s.target.sqt", g_staticParams.inputFile.szBaseName);
+             #else
              sprintf(szOutputSQT, "%s.sqt", g_staticParams.inputFile.szBaseName);
+             #endif
           else
              sprintf(szOutputSQT, "%s.%d-%d.sqt", g_staticParams.inputFile.szBaseName, iFirstScan, iLastScan);
 
           if ((fpout_sqt = fopen(szOutputSQT, "w")) == NULL)
           {
-             fprintf(stderr, "Error - cannot write to file \"%s\".\n\n", szOutputSQT);
+             logerr("Error - cannot write to file \"%s\".\n\n", szOutputSQT);
              exit(1);
           }
 
@@ -175,7 +179,7 @@
 
              if ((fpoutd_sqt = fopen(szOutputDecoySQT, "w")) == NULL)
              {
-                fprintf(stderr, "Error - cannot write to decoy file \"%s\".\n\n", szOutputDecoySQT);
+                logerr("Error - cannot write to decoy file \"%s\".\n\n", szOutputDecoySQT);
                 exit(1);
              }
           }
@@ -184,13 +188,17 @@
        if (g_staticParams.options.bOutputTxtFile)
        {
           if (iAnalysisType == AnalysisType_EntireFile)
+             #ifdef CRUX
+             sprintf(szOutputTxt, "%s.target.txt", g_staticParams.inputFile.szBaseName);
+             #else
              sprintf(szOutputTxt, "%s.txt", g_staticParams.inputFile.szBaseName);
+             #endif
           else
              sprintf(szOutputTxt, "%s.%d-%d.txt", g_staticParams.inputFile.szBaseName, iFirstScan, iLastScan);
 
           if ((fpout_txt = fopen(szOutputTxt, "w")) == NULL)
           {
-             fprintf(stderr, "Error - cannot write to file \"%s\".\n\n", szOutputTxt);
+             logerr("Error - cannot write to file \"%s\".\n\n", szOutputTxt);
              exit(1);
           }
 
@@ -203,7 +211,7 @@
 
              if ((fpoutd_txt= fopen(szOutputDecoyTxt, "w")) == NULL)
              {
-                fprintf(stderr, "Error - cannot write to decoy file \"%s\".\n\n", szOutputDecoyTxt);
+                logerr("Error - cannot write to decoy file \"%s\".\n\n", szOutputDecoyTxt);
                 exit(1);
              }
           }
@@ -212,13 +220,17 @@
        if (g_staticParams.options.bOutputPepXMLFile)
        {
           if (iAnalysisType == AnalysisType_EntireFile)
+             #ifdef CRUX
+             sprintf(szOutputPepXML, "%s.target.pep.xml", g_staticParams.inputFile.szBaseName);
+             #else
              sprintf(szOutputPepXML, "%s.pep.xml", g_staticParams.inputFile.szBaseName);
+             #endif
           else
              sprintf(szOutputPepXML, "%s.%d-%d.pep.xml", g_staticParams.inputFile.szBaseName, iFirstScan, iLastScan);
 
           if ((fpout_pepxml = fopen(szOutputPepXML, "w")) == NULL)
           {
-             fprintf(stderr, "Error - cannot write to file \"%s\".\n\n", szOutputPepXML);
+             logerr("Error - cannot write to file \"%s\".\n\n", szOutputPepXML);
              exit(1);
           }
 
@@ -234,7 +246,7 @@
 
              if ((fpoutd_pepxml = fopen(szOutputDecoyPepXML, "w")) == NULL)
              {
-                fprintf(stderr, "Error - cannot write to decoy file \"%s\".\n\n", szOutputDecoyPepXML);
+                logerr("Error - cannot write to decoy file \"%s\".\n\n", szOutputDecoyPepXML);
                 exit(1);
              }
 
@@ -246,13 +258,17 @@
        if (g_staticParams.options.bOutputPinXMLFile)
        {
           if (iAnalysisType == AnalysisType_EntireFile)
+             #ifdef CRUX
+             sprintf(szOutputPinXML, "%s.target.pin.xml", g_staticParams.inputFile.szBaseName);
+             #else
              sprintf(szOutputPinXML, "%s.pin.xml", g_staticParams.inputFile.szBaseName);
+             #endif
           else
              sprintf(szOutputPinXML, "%s.%d-%d.pin.xml", g_staticParams.inputFile.szBaseName, iFirstScan, iLastScan);
 
           if ((fpout_pinxml = fopen(szOutputPinXML, "w")) == NULL)
           {
-             fprintf(stderr, "Error - cannot write to file \"%s\".\n\n", szOutputPinXML);
+             logerr("Error - cannot write to file \"%s\".\n\n", szOutputPinXML);
              exit(1);
           }
 
@@ -277,7 +293,7 @@
        {
           // Load and preprocess all the spectra.
           if (!g_staticParams.options.bOutputSqtStream)
-             printf(" - Load and process input spectra\n");
+             logout(" - Load and process input spectra\n");
 
           CometPreprocess::LoadAndPreprocessSpectra(mstReader,
                 iFirstScan, iLastScan, iAnalysisType,
@@ -291,12 +307,12 @@
 
           // Allocate memory to store results for each query spectrum.
           if (!g_staticParams.options.bOutputSqtStream)
-             printf(" - Allocate memory to store results\n");
+             logout(" - Allocate memory to store results\n");
 
           AllocateResultsMem();
 
           if (!g_staticParams.options.bOutputSqtStream)
-             printf(" - Number of mass-charge spectra loaded: %d\n", (int)g_pvQuery.size());
+             logout(" - Number of mass-charge spectra loaded: %d\n", (int)g_pvQuery.size());
 
           // Sort g_pvQuery vector by dExpPepMass.
           std::sort(g_pvQuery.begin(), g_pvQuery.end(), compareByPeptideMass);
@@ -313,7 +329,7 @@
           CalcRunTime(tStartTime);
 
           if (!g_staticParams.options.bOutputSqtStream)
-             printf(" - Write output\n");
+             logout(" - Write output\n");
 
           if (g_staticParams.options.bOutputOutFiles)
              CometWriteOut::WriteOut();
@@ -341,14 +357,14 @@
        }
        if (iTotalSpectraSearched == 0)
        {
-          printf(" Warning - no spectra searched.\n\n");
+          logout(" Warning - no spectra searched.\n\n");
        }
 
        if (!g_staticParams.options.bOutputSqtStream)
        {
           time(&tStartTime);
           strftime(g_staticParams.szDate, 26, "%m/%d/%Y, %I:%M:%S %p", localtime(&tStartTime));
-          printf(" Search end:    %s\n\n", g_staticParams.szDate);
+          logout(" Search end:    %s\n\n", g_staticParams.szDate);
        }
 
        if (NULL != fpout_pepxml)
@@ -444,7 +460,7 @@
 
          if (err != EEXIST) 
          {
-            fprintf(stderr, "\n Error - could not create directory \"%s\".\n", g_staticParams.inputFile.szBaseName);
+            logerr("\n Error - could not create directory \"%s\".\n", g_staticParams.inputFile.szBaseName);
             exit(1);
          }
       }
@@ -460,7 +476,7 @@
 
             if (err != EEXIST) 
             {
-               fprintf(stderr, "\n Error - could not create directory \"%s\".\n", szDecoyDir);
+               logerr("\n Error - could not create directory \"%s\".\n", szDecoyDir);
                exit(1);
             }
          }
@@ -468,7 +484,7 @@
 #else
       if ((mkdir(g_staticParams.inputFile.szBaseName, 0775) == -1) && (errno != EEXIST))
       {
-         fprintf(stderr, "\n Error - could not create directory \"%s\".\n", g_staticParams.inputFile.szBaseName);
+         logerr("\n Error - could not create directory \"%s\".\n", g_staticParams.inputFile.szBaseName);
          exit(1);
       }
       if (g_staticParams.options.iDecoySearch == 2)
@@ -478,7 +494,7 @@
 
          if ((mkdir(szDecoyDir , 0775) == -1) && (errno != EEXIST))
          {
-            fprintf(stderr, "\n Error - could not create directory \"%s\".\n\n", szDecoyDir);
+            logerr("\n Error - could not create directory \"%s\".\n\n", szDecoyDir);
             exit(1);
          }
       }
@@ -511,7 +527,7 @@
 
       if (pQuery->_pResults == NULL)
       {
-         fprintf(stderr, " Error malloc(_pResults[])\n");
+         logerr(" Error malloc(_pResults[])\n");
          exit(1);
       }
 
@@ -529,7 +545,7 @@
 
          if (pQuery->_pDecoys == NULL)
          {
-            fprintf(stderr, " Error malloc(_pDecoys[])\n");
+            logerr(" Error malloc(_pDecoys[])\n");
             exit(1);
          }
 
Index: CometSearch/CometWriteOut.cpp
===================================================================
--- CometSearch/CometWriteOut.cpp	(revision 131)
+++ CometSearch/CometWriteOut.cpp	(working copy)
@@ -139,7 +139,7 @@
 
    if ((fpout = fopen(szOutput, "w")) == NULL)
    {
-      fprintf(stderr, "Error - cannot write to file %s.\n\n", szOutput);
+      logerr( "Error - cannot write to file %s.\n\n", szOutput);
       exit(1);
    }
 
Index: CometSearch/CometWritePepXML.cpp
===================================================================
--- CometSearch/CometWritePepXML.cpp	(revision 131)
+++ CometSearch/CometWritePepXML.cpp	(working copy)
@@ -98,7 +98,7 @@
    // Grab file extension from file name
    if ( (pStr2 = strrchr(g_staticParams.inputFile.szFileName, '.')) == NULL)
    {
-      printf(" Error - in WriteXMLHeader missing last period in file name: %s\n", g_staticParams.inputFile.szFileName);
+      logout(" Error - in WriteXMLHeader missing last period in file name: %s\n", g_staticParams.inputFile.szFileName);
       exit(1);
    }
    pStr2++;
@@ -126,7 +126,7 @@
    char szParamBuf[SIZE_BUF];
    if ((fp=fopen(szParamsFile, "r")) == NULL)
    {
-      fprintf(stderr, " Error - cannot open parameter file %s.\n\n", szParamsFile);
+      logerr(" Error - cannot open parameter file %s.\n\n", szParamsFile);
       exit(1);
    }
 
Index: CometSearch/CometWritePinXML.cpp
===================================================================
--- CometSearch/CometWritePinXML.cpp	(revision 131)
+++ CometSearch/CometWritePinXML.cpp	(working copy)
@@ -18,8 +18,8 @@
 #include "CometData.h"
 #include "CometMassSpecUtils.h"
 #include "CometWritePinXML.h"
+#include <math.h>
 
-
 CometWritePinXML::CometWritePinXML()
 {
 }
@@ -152,7 +152,7 @@
    }
    else
    {
-      fprintf(stderr, " Error, dMZ=0. scan %d\n", pQuery->_spectrumInfoInternal.iScanNumber);
+      logerr(" Error, dMZ=0. scan %d\n", pQuery->_spectrumInfoInternal.iScanNumber);
       exit(1);
    }
 
Index: CometSearch/CometWriteSqt.cpp
===================================================================
--- CometSearch/CometWriteSqt.cpp	(revision 131)
+++ CometSearch/CometWriteSqt.cpp	(working copy)
@@ -94,7 +94,7 @@
    // write out entire parameters file to SQT header
    if ((fp=fopen(szParamsFile, "r")) == NULL)
    {
-      fprintf(stderr, " Error - cannot open parameter file %s.\n\n", szParamsFile);
+      logerr(" Error - cannot open parameter file %s.\n\n", szParamsFile);
       exit(1);
    }
 
Index: CometSearch/Common.h
===================================================================
--- CometSearch/Common.h	(revision 131)
+++ CometSearch/Common.h	(working copy)
@@ -42,6 +42,11 @@
 #include "MSObject.h"
 #include <vector>
 
+#ifdef CRUX
+#include <iostream>
+#endif
+
+
 #define comet_version   "2013.01 rev. 0"
 #define copyright "(c) University of Washington"
 
@@ -49,4 +54,13 @@
 // now between 0.0 and 1.0 and scales to the binWidth.
 #define BIN(dMass) (int)(dMass*g_staticParams.dInverseBinWidth + g_staticParams.dOneMinusBinOffset)
 
+using namespace MSToolkit;
+#ifdef CRUX
+#define logout(...) {char sbuf[1000]; snprintf(sbuf, 1000, __VA_ARGS__); cerr << sbuf;}
+#define logerr(...) {char sbuf[1000]; snprintf(sbuf, 1000, __VA_ARGS__); cerr << sbuf;}
+#else
+#define logout(...) printf(__VA_ARGS__)
+#define logerr(...) fprintf(stderr, __VA_ARGS__)
+#endif
+
 #endif // _COMMON_H_
Index: MSToolkit/include/H5pubconf.h
===================================================================
--- MSToolkit/include/H5pubconf.h	(revision 131)
+++ MSToolkit/include/H5pubconf.h	(working copy)
@@ -74,7 +74,7 @@
 #define H5_HAVE_C99_FUNC 1
 
 /* Define to 1 if you have the `clock_gettime' function. */
-#define H5_HAVE_CLOCK_GETTIME 1
+/* #undef H5_HAVE_CLOCK_GETTIME */
 
 /* Define if the function stack tracing code is to be compiled in */
 /* #undef H5_HAVE_CODESTACK */
@@ -99,7 +99,7 @@
 #define H5_HAVE_EMBEDDED_LIBINFO 1
 
 /* Define to 1 if you have the <features.h> header file. */
-#define H5_HAVE_FEATURES_H 1
+/* #undef H5_HAVE_FEATURES_H */
 
 /* Define if support for deflate (zlib) filter is enabled */
 #define H5_HAVE_FILTER_DEFLATE 1
@@ -132,7 +132,7 @@
 #define H5_HAVE_FSEEKO 1
 
 /* Define to 1 if you have the `fseeko64' function. */
-#define H5_HAVE_FSEEKO64 1
+/* #undef H5_HAVE_FSEEKO64 */
 
 /* Define to 1 if you have the `fstat64' function. */
 #define H5_HAVE_FSTAT64 1
@@ -141,10 +141,10 @@
 #define H5_HAVE_FTELLO 1
 
 /* Define to 1 if you have the `ftello64' function. */
-#define H5_HAVE_FTELLO64 1
+/* #undef H5_HAVE_FTELLO64 */
 
 /* Define to 1 if you have the `ftruncate64' function. */
-#define H5_HAVE_FTRUNCATE64 1
+/* #undef H5_HAVE_FTRUNCATE64 */
 
 /* Define if the compiler understands the __FUNCTION__ keyword */
 #define H5_HAVE_FUNCTION 1
@@ -229,13 +229,13 @@
 #define H5_HAVE_LONGJMP 1
 
 /* Define to 1 if you have the `lseek64' function. */
-#define H5_HAVE_LSEEK64 1
+/* #undef H5_HAVE_LSEEK64 */
 
 /* Define to 1 if you have the `lstat' function. */
 #define H5_HAVE_LSTAT 1
 
 /* Define to 1 if you have the <mach/mach_time.h> header file. */
-/* #undef H5_HAVE_MACH_MACH_TIME_H */
+#define H5_HAVE_MACH_MACH_TIME_H 1
 
 /* Define to 1 if you have the <memory.h> header file. */
 #define H5_HAVE_MEMORY_H 1
@@ -286,7 +286,7 @@
 #define H5_HAVE_SIGPROCMASK 1
 
 /* Define to 1 if you have the `sigsetjmp' function. */
-/* #undef H5_HAVE_SIGSETJMP */
+#define H5_HAVE_SIGSETJMP 1
 
 /* Define to 1 if you have the `snprintf' function. */
 #define H5_HAVE_SNPRINTF 1
@@ -373,7 +373,7 @@
 /* #undef H5_HAVE_THREADSAFE */
 
 /* Define if `timezone' is a global variable */
-/* #undef H5_HAVE_TIMEZONE */
+#define H5_HAVE_TIMEZONE 1
 
 /* Define if the ioctl TIOCGETD is defined */
 #define H5_HAVE_TIOCGETD 1
@@ -534,10 +534,10 @@
 #define H5_SIZEOF_INT8_T 1
 
 /* The size of `int_fast16_t', as computed by sizeof. */
-#define H5_SIZEOF_INT_FAST16_T 8
+#define H5_SIZEOF_INT_FAST16_T 2
 
 /* The size of `int_fast32_t', as computed by sizeof. */
-#define H5_SIZEOF_INT_FAST32_T 8
+#define H5_SIZEOF_INT_FAST32_T 4
 
 /* The size of `int_fast64_t', as computed by sizeof. */
 #define H5_SIZEOF_INT_FAST64_T 8
@@ -567,7 +567,7 @@
 #define H5_SIZEOF_LONG_LONG 8
 
 /* The size of `off64_t', as computed by sizeof. */
-#define H5_SIZEOF_OFF64_T 8
+#define H5_SIZEOF_OFF64_T 0
 
 /* The size of `off_t', as computed by sizeof. */
 #define H5_SIZEOF_OFF_T 8
@@ -597,10 +597,10 @@
 #define H5_SIZEOF_UINT8_T 1
 
 /* The size of `uint_fast16_t', as computed by sizeof. */
-#define H5_SIZEOF_UINT_FAST16_T 8
+#define H5_SIZEOF_UINT_FAST16_T 2
 
 /* The size of `uint_fast32_t', as computed by sizeof. */
-#define H5_SIZEOF_UINT_FAST32_T 8
+#define H5_SIZEOF_UINT_FAST32_T 4
 
 /* The size of `uint_fast64_t', as computed by sizeof. */
 #define H5_SIZEOF_UINT_FAST64_T 8
Index: MSToolkit/include/MSObject.h
===================================================================
--- MSToolkit/include/MSObject.h	(revision 131)
+++ MSToolkit/include/MSObject.h	(working copy)
@@ -8,6 +8,7 @@
 
 using namespace std;
 
+namespace MSToolkit {
 class MSObject {
  public:
   //Constructors & Destructors
@@ -40,5 +41,6 @@
   
 };
 
+}
 #endif
 
Index: MSToolkit/include/MSReader.h
===================================================================
--- MSToolkit/include/MSReader.h	(revision 131)
+++ MSToolkit/include/MSReader.h	(working copy)
@@ -53,6 +53,7 @@
 
 using namespace std;
 
+namespace MSToolkit {
 class MSReader {
  public:
   //Constructors & Destructors
@@ -173,5 +174,6 @@
 
 };
 
+}
 #endif
 
Index: MSToolkit/include/MSToolkitTypes.h
===================================================================
--- MSToolkit/include/MSToolkitTypes.h	(revision 131)
+++ MSToolkit/include/MSToolkitTypes.h	(working copy)
@@ -2,6 +2,7 @@
 #define _MSTOOLKITTYPES_H
 
 #include <stddef.h>
+namespace MSToolkit {
 
 enum MSSpectrumType {
   MS1,
@@ -114,6 +115,8 @@
   float pArea;    //precursor retention time
 };
 
+}
+
 #endif
 
 
Index: MSToolkit/include/Spectrum.h
===================================================================
--- MSToolkit/include/Spectrum.h	(revision 131)
+++ MSToolkit/include/Spectrum.h	(working copy)
@@ -9,6 +9,7 @@
 
 using namespace std;
 
+namespace MSToolkit {
 class Spectrum {
  public:
   //Constructors & Destructors
@@ -148,6 +149,7 @@
 
 };
 
+}
 
 #endif
 
Index: MSToolkit/src/MSToolkit/MSObject.cpp
===================================================================
--- MSToolkit/src/MSToolkit/MSObject.cpp	(revision 131)
+++ MSToolkit/src/MSToolkit/MSObject.cpp	(working copy)
@@ -3,6 +3,8 @@
 
 using namespace std;
 
+namespace MSToolkit {
+
 MSObject::MSObject(){
   vSpectrum = new vector<Spectrum>;
   fileName="";
@@ -96,3 +98,4 @@
   return vSpectrum->size();
 };
 
+}
Index: MSToolkit/src/MSToolkit/MSReader.cpp
===================================================================
--- MSToolkit/src/MSToolkit/MSReader.cpp	(revision 131)
+++ MSToolkit/src/MSToolkit/MSReader.cpp	(working copy)
@@ -2,6 +2,8 @@
 #include <iostream>
 using namespace std;
 
+namespace MSToolkit {
+
 MSReader::MSReader(){
   fileIn=NULL;
   rampFileIn=NULL;
@@ -1976,3 +1978,4 @@
 
 }
 
+}
Index: MSToolkit/src/MSToolkit/Spectrum.cpp
===================================================================
--- MSToolkit/src/MSToolkit/Spectrum.cpp	(revision 131)
+++ MSToolkit/src/MSToolkit/Spectrum.cpp	(working copy)
@@ -4,6 +4,7 @@
 
 using namespace std;
 
+namespace MSToolkit {
 Spectrum::Spectrum(){
   //cout<<"in Spectrum constructor!"<<endl;
 
@@ -569,3 +570,4 @@
   else return 0;
 }
 
+}
Index: Makefile
===================================================================
--- Makefile	(revision 131)
+++ Makefile	(working copy)
@@ -1,7 +1,7 @@
 CXX = g++
 MSTOOLKIT = MSToolkit
 COMETSEARCH = CometSearch
-CXXFLAGS = -O3 -static -Wall -Wno-char-subscripts -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -I$(MSTOOLKIT)/include -I$(COMETSEARCH)
+CXXFLAGS = -O3 -Wall -Wno-char-subscripts -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -I$(MSTOOLKIT)/include -I$(COMETSEARCH)
 EXECNAME = comet.exe
 OBJS = Comet.o
 DEPS = 
